types/plan/plan.go

```diff
@@ -177,15 +178,57 @@ func (output *KubernetesOutput) Merge(newoutput KubernetesOutput) {
 
 // Inputs defines the input section of plan
 type Inputs struct {
+	AbsRootDir          string                                   `yaml:"absRootDir"`
+	RelRootDir          string                                   `yaml:"relRootDir"`
 	RootDir             string                                   `yaml:"rootDir"`
 	K8sFiles            []string                                 `yaml:"kubernetesYamls,omitempty"`
 	QACaches            []string                                 `yaml:"qaCaches,omitempty"`
 	Services            map[string][]Service                     `yaml:"services"`                      // [serviceName][Services]
 	TargetInfoArtifacts map[TargetInfoArtifactTypeValue][]string `yaml:"targetInfoArtifacts,omitempty"` //[targetinfoartifacttype][List of artifacts]
 }
 
+// SetRootDir sets the path to the source directory in the plan.Spec.Inputs
+func (inputs *Inputs) SetRootDir(path string) error {
+	path = filepath.Clean(path)
+	inputs.RootDir = path
+
+	if filepath.IsAbs(path) {
+		inputs.AbsRootDir = path
+		currDir, err := os.Getwd()
+		if err != nil {
+			log.Errorf("Failed to get the current working directory. Error %q", err)
+			return err
+		}
+		relPath, err := filepath.Rel(currDir, path)
+		if err != nil {
+			log.Errorf("Failed to make the source directory %q relative to the current working directory %q Error %q", path, currDir, err)
+			return err
+		}
+		inputs.RelRootDir = relPath
+	} else {
+		inputs.RelRootDir = path
+		absPath, err := filepath.Abs(path)
+		if err != nil {
+			log.Errorf("Failed to get the absolute path for the source directory %q Error %q", path, err)
+			return err
+		}
+		inputs.AbsRootDir = absPath
+	}
+
+	return nil
+}
+
+// RepoInfo contains information specific to creating the CI/CD pipeline.
+type RepoInfo struct {
+	GitRepoDir    string
+	GitRepoURL    string
+	GitRepoBranch string
+	TargetPath    string
+}
+
 // Service defines a plan service
 type Service struct {
+	RepoInfo                      RepoInfo
```

Comment:
We need to add serialization tags for this. and each internal artifact.

Reply:
Also, I think we need not have repoinfo in plan. It can be present only in the IR if we don't have any plans of editing it anyway. The goal here is to keep the plan crisp showing only the critical info.

Reply:
`We need to add serialization tags for this. and each internal artifact.` Done